### PR TITLE
Migrate quote e2e tests to playwright

### PR DESCRIPTION
**Pull request by t-hamano**
## What?
Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate `quote.test.js` to its Playwright version.

## Why?

Part of #38851.

## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.

## Testing Instructions

Run `npm run test:e2e:playwright test/e2e/specs/editor/blocks/quote.spec.js`

## Note

- The quart block already had one playrwight test. I migrated the existing Pupetter test against that test file.
- In blocks already migrated to Playright, both inline strings and snapshots are used in the matcher. I couldn't find a clear rule as to which one to use, but I used snapshots for all tests.
- As reported in #46519, the quart block test is unstable and I expect that this will also be resolved by rewriting to Playwright.

### CODE REVIEW

File: test/e2e/specs/editor/blocks/quote.spec.js [hunk 1]
```diff
@@ -3,30 +3,189 @@
...
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/quote' } );
+		await page.keyboard.type( '1' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( '2' );
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+		await pageUtils.pressKeyTimes( 'ArrowLeft', 4 );
```
Comment (t-hamano) [inline test/e2e/specs/editor/blocks/quote.spec.js:308]: Originally,[ the arrow-left was used for keystrokes](https://github.com/WordPress/gutenberg/blob/3245d1e0518d6a2fa565266981fe0fab8a984a8e/packages/e2e-tests/specs/editor/blocks/quote.test.js#L185-L186).

However, as discovered in [this comment](https://github.com/WordPress/gutenberg/issues/46519#issuecomment-1351644708), the arrow-up key may cause unstable behavior depending on the theme, so it is rewritten as the arrow-left key.

**Comment by Mamaduka** [comment]
Thanks for working on this, @t-hamano 🦸

The migration looks great, and I will review this first thing tomorrow 👍

> In blocks already migrated to Playright, both inline strings and snapshots are used in the matcher. I couldn't find a clear rule as to which one to use, but I used snapshots for all tests.

[The updated guidelines](https://github.com/WordPress/gutenberg/pull/45709/) haven't been merged yet, but the inline snapshot would be preferred here. Plus, we don't have a block assertions helper in place yet. See the rendered document about snapshot assertions: https://github.com/WordPress/gutenberg/blob/f2d2fa14f18ba1d257ede08b714940d4c3ba5590/docs/contributors/code/e2e/overusing-snapshots.md.

**Comment by t-hamano** [comment]
Thanks for the info, @Mamaduka!

I have read [the update guidelines](https://github.com/WordPress/gutenberg/pull/45709/). Personally, I think inline snapshots are appropriate for this test, but if the approach using `toMatchObject` and the new `getBlocks` utility is called for, I would be happy to work.

**Comment by Mamaduka** [comment]
@t-hamano, using inline snapshots with `.toBe()` should be good for now 👍

**Comment by t-hamano** [comment]
I replaced it with an inline snapshot.
So far, no test failures seem to have occurred.

File: test/e2e/specs/editor/blocks/quote.spec.js [hunk 2]
```diff
@@ -3,29 +3,310 @@
  */
 const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
 
-test.describe( 'adding a quote', () => {
+test.describe( 'Quote', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
```
Comment (Mamaduka) [inline test/e2e/specs/editor/blocks/quote.spec.js:10]: We should probably clean up after all tests are done.

```js
test.afterAll( async ( { requestUtils } ) => {
  await requestUtils.deleteAllPosts();
} );
```

File: test/e2e/specs/editor/blocks/quote.spec.js [hunk 3]
```diff
@@ -3,29 +3,310 @@
...
+	test( 'can be created by converting multiple paragraphs', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'one' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'two' );
+		await page.keyboard.down( 'Shift' );
+		await page.click( '[data-type="core/paragraph"] >> nth=0' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/blocks/quote.spec.js]: Suggestion: You can use role selector + explicit text for the block. Maybe something like this:

```js
await page.click( 'role=document[name="Paragraph block"i] >> text=one' );
```

P.S. I've tested this selector in this case, but have used similar one before :)

File: test/e2e/specs/editor/blocks/quote.spec.js (hunk 1 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/blocks/quote.spec.js:308]: Maybe we should leave an inline comment here. What do you think?

**Review by Mamaduka — APPROVED** [review]
Thank you, @t-hamano! The migration looks great.

I've only left a few suggestions.

File: test/e2e/specs/editor/blocks/quote.spec.js (hunk 2 above)
Comment (t-hamano) [reply test/e2e/specs/editor/blocks/quote.spec.js:10]: Done.
By the way, I think that once #46459 is merged, The clean-up might no longer be needed 🤔

File: test/e2e/specs/editor/blocks/quote.spec.js (hunk 3 above)
Comment (t-hamano) [reply test/e2e/specs/editor/blocks/quote.spec.js]: Done.

**Comment by t-hamano** [comment]
> Maybe we should leave an inline comment here. What do you think?

I have added the following comment, is it appropriate?

```javascript
// Move the cursor to the start of the first paragraph of the quoted block.
await pageUtils.pressKeyTimes( 'ArrowLeft', 4 );
```

File: test/e2e/specs/editor/blocks/quote.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/blocks/quote.spec.js:10]: Thanks. That would be great. We can remove post-cleanup methods after that is confirmed in #46459.

**Comment by Mamaduka** [comment]
Thanks again. Feel free to merge after all checks are green ✅

**Pull request merged**